comp/api/api/apiimpl/observability/utils_test.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package observability
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestExtractPath(t *testing.T) {
+	testCases := []struct {
+		path     string
+		expected string
+	}{
+		{"/", "/"},
+		{"/test", "/test"},
+		{"/test/", "/test/"},
+		{"/test/1", "/test/1"},
+		{"/test/1?arg=value", "/test/1"},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.path, func(t *testing.T) {
+			req := httptest.NewRequest("GET", "http://agent.host"+tc.path, nil)
+			assert.Equal(t, tc.expected, extractPath(req))
+		})
+	}
+
+	t.Run("invalid url", func(t *testing.T) {
+		req := http.Request{RequestURI: "http://agent.host/invalidurl" + string([]byte{0x7f})}
+		assert.Equal(t, "<invalid url>", extractPath(&req))
+	})
+}
+
+func TestExtractStatusHandler(t *testing.T) {
+	// can't test with status code 1xx since they are not final responses
+	testCases := []int{
+		http.StatusOK, http.StatusMovedPermanently, http.StatusNotFound, http.StatusInternalServerError,
+	}
+
+	for _, tcStatus := range testCases {
+		t.Run(http.StatusText(tcStatus), func(t *testing.T) {
+			var status int
```

Comment:
Should we check the value of the status variable, it seems the `extractStatusHandler` modifies it.

Reply:
That's what is done below (line 66), it needs to happen after the request took place

Reply:
Ohh. I'm confused then.

On line 66 we assert `assert.Equal(t, tcStatus, resp.StatusCode)` we are comparing the response code with the value from the test cases, but I do not see any assertion against `var status` 

By the look of it, when we use this middleware in production the value that we use as the status code is the value store in the int pointer passed to the middleware, that is why I thought we should test against that value as well.

```go
var statusCode int
next = extractStatusCodeHandler(&statusCode)(next)

var duration time.Duration
next = timeHandler(clock, &duration)(next)

next.ServeHTTP(w, r)

logFunc := getLogFunc(statusCode)
path := extractPath(r)
logFunc(logFormat, serverName, r.Method, path, r.RemoteAddr, r.Host, duration, statusCode)
```

Reply:
I actually misread the code, you are right the current version asserts the values of `resp.StatusCode`, while it should assert `statusCode`, I fixed that !